tests/e2e/scenarios/test_reborn_webui_v2_tool_gates.py

```
@@ -0,0 +1,350 @@
+
+    while asyncio.get_running_loop().time() < deadline:
+        remaining = deadline - asyncio.get_running_loop().time()
+        try:
+            raw = await asyncio.wait_for(response.content.readline(), timeout=remaining)
+        except asyncio.TimeoutError:
+            break
+        if not raw:
+            raise AssertionError(
+                "SSE stream closed before a matching event arrived; "
+                f"wanted={event_types}, seen={seen_events}"
+            )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)